### DISCUSSION TITLE
BTP Blocks

### DISCUSSION THREAD

**Discussion by izyak** [icon-project/btp#152] [General]
We opened a BTP Network on height `0x1a` and the following eventlog was generated.

```py
{
      "scoreAddress": "cx0000000000000000000000000000000000000000",
      "indexed": [
        "BTPNetworkOpened(int,int)",
        "0x1",
        "0x1"
      ],
      "data": []
}
```

Then, we sent a BTP Message at height `0x1f` which generated the following eventlog.

```py
{
      "scoreAddress": "cx0000000000000000000000000000000000000000",
      "indexed": [
        "BTPMessage(int,int)",
        "0x1",
        "0x0"
      ],
      "data": []
    }

```
When we try to get `btp_getProof` or `btp_getMessage` for network 0x1 and height `0x1f`, we get empty data, but we get some data on height `0x20`. 

Similarly, with `icx_getProofForEvents` as well, which takes block hash as param, we had to provide block hash for block `0x20`

Is this behavior expected?

**Comment by mksong76** [comment]
Yes, it's expected behavior.
The transaction is added to N block, then the result would be recorded in N+1 block.

**Comment by BennyOptions** [comment]
cc @CyrusVorwald for knowledge-sharing purposes
